burnman/seismic.py

```diff
@@ -229,38 +282,46 @@ def density(self, depth):
 
     def depth(self, pressure):
         if pressure > max(self.table_pressure) or pressure < min(self.table_pressure)  :
-           raise ValueError, "Pressure outside range of PREM"
-        radius = np.interp(pressure, self.table_pressure[::-1], self.table_radius[::-1] )
-        return self.earth_radius - radius
+            raise ValueError, "Pressure outside range of SeismicRadiusTable"
+        if not len(self.table_depth) == 0:
+            depth = np.interp(pressure, self.table_pressure, self.table_depth )
+        else:
+            depth = np.interp(pressure, self.table_pressure[::-1], self.earth_radius - self.table_radius[::-1])
```

Comment:
Not needed if you do everything in radius, but I was suggesting something like
```
   depth =  self.earth_radius - np.interp(pressure, self.table_pressure,self.table_radius)
```
so you don't create another temporary array